ci/repo_cleanups/remove_stale_rpm.sh

```diff
@@ -0,0 +1,231 @@
+                *)
+                echo 0
+        esac
+    
+    }
+   
+    function get_doy {
+        yyyymmdd=$1
+
+        dt=${yyyymmdd:0:8}
+        date -d "${dt}"  +"%_j"
+
+    }
+
+    function today {
+        date +"%Y%m%d"
+    }
+
+    # Compare two dates and subtract one (the first arg.) from the other. Answer is given in days.
+    function delta_dates {
+        ## The date formate must be: yyyymmdd
+        date1=$1
+        date2=$2
+
+        yr1=${date1:0:4}
+        yr2=${date2:0:4}
+
+        doy1=$(get_doy ${date1})
+        doy2=$(get_doy ${date2})
+
+        deltaY=$(($yr2 - $yr1))
+        days_to_add=$((365*${deltaY}))   ## Ignore leap year extra day!
+
+        echo $(( ${doy2} - ${doy1} + ${days_to_add}))
+
+    }
+
+ 
+    ###################################################################################
+    # Get the list (has) of git branches that are still "live"
+    for key in `git_branch_info`; do
+       git_branches[$key]=1
+    done
+    
+    
+    origin=$PWD
+    cd ${rpm_repo_dir}
+    for dr in $(find ./* -maxdepth 0 -type d); do
+       dr=${dr##*\/}
+    
+       if [[ "$dr" = "." ]]; then
+           continue            ## Ignore the current directory
+       fi
+    
+       branch_name=${git_branches[$dr]}
+       if [[ "${branch_name}" = "master" ]]; then
+           continue
+       fi
+       if [[ "${branch_name}" = "develop" ]]; then
+           continue
+       fi
+   
+       ## If the branch name is *not* in the git_branches hash, remove it! 
+       if [[ -z ${branch_name} ]]; then
+            remove_dir ${dr}
+            if [ $? -ne 0 ]; then
+                exit 1
+            fi
+       fi
+    
+    done
+    
+    ## Now delete superfluous rpm's from the master directory
+    cd master
+    if [[ $? -ne 0 ]]; then
+        exit 0                # There's a problem here, but we won't worry about it 
+    fi
+   
+    nrepos=$(ls -1 . | wc -l)
+    if [[ ${nrepos} -lt 2 ]]; then
+        echo "Something is wrong. The master directory contains one or less repos. Quitting."
+        exit 1
+    fi
+ 
+    current=$(readlink current)
+    if [[ -z ${current} ]]; then
+        echo "No 'current' symlink in master! "
+        exit 1                # There is no "current" symlink. Don't remove anything!
+    fi
+    echo "'current' rpm repo is ${current}"
+
+    ## Just to be safe! We'll do the same 
+    readlink current
+    if [[ $? -ne 0 ]]; then
+        echo "No 'current' symlink in master! "
+        exit 1
+    fi
+
+    current=${current##*\/}
+    now=$(today)
+    echo "Checking for stale rpm repos under master..."
+    for dt in $(ls -d 2*); do
+        rpmdate=${dt:0:8}     # yyyymmdd is the format
+
+        ndays=$(delta_dates ${rpmdate} ${now})
+
+        if [[ "${dt}" = "${current}" ]]; then
+            continue
+        fi
+
+        if [[ ${ndays} -gt 14 ]]; then
+            ## Just in case the continue command does not work as I think it should...
```

Comment:
What is the concern you have with the continue command? The way it works is documented well in the [bash man page](https://linux.die.net/man/1/bash).

> continue [n]
Resume the next iteration of the enclosing for, while, until, or select loop. If n is specified, resume at the nth enclosing loop. n must be ≥ 1. If n is greater than the number of enclosing loops, the last enclosing loop (the ''top-level'' loop) is resumed. When continue is executed inside of loop, the return value is non-zero when n is ≤ 0; Otherwise, continue returns 0 value. When continue is executed outside of loop, the return value is 0.

I also used to write code like this but I think it's better to spend a little extra time and understand exactly what the commands you use are doing. That way there's no need for clauses like this and it simplifies the code.